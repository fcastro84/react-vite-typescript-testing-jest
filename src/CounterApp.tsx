import { useState } from "react"

interface CounterProps {
    value: number
}

const CounterApp = ({value}: CounterProps) => {

    const [counter, setCounter] = useState(value)

    const handleAdd = () => {
        setCounter( counter + 1 )
    }

    const handleSubs = () => {
        setCounter( counter - 1 )
    }

    const handleReset = () => {
        setCounter( value )
    }

  return (
    <div className="center">
        <h1>CounterApp</h1>
        <span>{ counter }</span>
        <div className="button">
            <button onClick={handleAdd}>+1</button>
            <button onClick={handleSubs}>-1</button>
            <button aria-label="btn-reset" onClick={handleReset}>Reset</button>
        </div>
      
    </div>
  )
}

export default CounterApp
