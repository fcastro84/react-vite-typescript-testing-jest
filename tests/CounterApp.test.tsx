 import { fireEvent, render, screen } from '@testing-library/react'
import CounterApp from '../src/CounterApp';
import React from 'react';


describe(' Test in component <CounterApp/>', () => {

    const value = 10

    test('This has make to match with the snapshot ', () => {

      const { container } = render(<CounterApp value={value} />)
      expect( container ).toMatchSnapshot()

    })

    test('this should show 100 as the initial value', () => {

        render(<CounterApp value={100} />)
        expect( screen.getByText(100) ).toBeTruthy()
        
      })

      test('this should add 1 with the button +1', () => {

        render(<CounterApp value={value} />)
        fireEvent.click( screen.getByText('+1') )
        expect( screen.getByText('11') ).toBeTruthy()
        
      })

      test('this should substract 1 with the button -1', () => {

        render(<CounterApp value={value} />)
        fireEvent.click( screen.getByText('-1') )
        // screen.debug()
        expect( screen.getByText('9') ).toBeTruthy()
        
      })

      test('the reset button should work', () => {

        render(<CounterApp value={value} />)
        fireEvent.click( screen.getByText('+1') )
        fireEvent.click( screen.getByText('+1') )
        fireEvent.click( screen.getByText('+1') )
         
        fireEvent.click( screen.getByRole('button', { name: 'btn-reset'}) )
        // fireEvent.click( screen.getByText('Reset') )
        // screen.debug()
        expect( screen.getByText(value) ).toBeTruthy()
        
      })
    


})