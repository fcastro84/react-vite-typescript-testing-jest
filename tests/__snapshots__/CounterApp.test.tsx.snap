// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Test in component <CounterApp/> This has make to match with the snapshot  1`] = `
<div>
  <div
    class="center"
  >
    <h1>
      CounterApp
    </h1>
    <span>
      10
    </span>
    <div
      class="button"
    >
      <button>
        +1
      </button>
      <button>
        -1
      </button>
      <button
        aria-label="btn-reset"
      >
        Reset
      </button>
    </div>
  </div>
</div>
`;
